test/cpp/api/functional.cpp

```diff
@@ -360,3 +360,24 @@ TEST_F(FunctionalTest, LogSigmoid) {
   auto y_exp = torch::log(torch::ones_like(x)/(torch::ones_like(x) + torch::exp(torch::neg(x))));
   ASSERT_TRUE(torch::allclose(y, y_exp, 1e-4, 1e-7));
 }
+
+TEST_F(FunctionalTest, Normalize) {
+  {
+    auto input = torch::randn({1,3,4,4}, torch::requires_grad());
+    auto norm = F::normalize(input, NormalizeOptions().p(1).dim(-1));
```

Comment:
It would be great to have specific values for `input` (e.g. `torch::arange(48).reshape({1, 3, 4, 4})`), and then check that the value of `norm` is what we expect it should be.

Reply:
I reduced the number of elements to 10 so that a readable tensor could be pasted into the test code. I can't think of another way of "generating" the solution on the fly without using the same steps used in the body of `F::normalize()` so I pre-calculated the solution using the Python API, is that considered acceptable?

My local source is still building, let's see if CI beats me to it :D

Reply:
Yes using pre-calculated solution from Python API is awesome. Thanks a lot for the changes! :D